File: src/main/resources/archetype-resources/fabric/src/main/java/__modName__Fabric.java
package ${package};

import me.shedaniel.autoconfig.AutoConfig;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.loader.api.FabricLoader;

import net.minecraft.client.gui.screens.Screen;

public final class ${modName}Fabric extends ${modName} implements ModInitializer {

    @Override
    public void onInitialize() {
    }

    @Override
    public boolean isClothConfigLoaded() {
        return FabricLoader.getInstance().isModLoaded("cloth-config2");
    }

    @Override
    public Screen getConfigScreen(final Screen parent) {
        return AutoConfig.getConfigScreen(${modName}ConfigData.java, parent).get();
    }
}

File: src/main/resources/archetype-resources/fabric/src/main/kotlin/__modName__Fabric.kt
package ${package}

import me.shedaniel.autoconfig.AutoConfig

import net.fabricmc.api.ModInitializer
import net.fabricmc.loader.api.FabricLoader

import net.minecraft.client.gui.screens.Screen

object ${modName}Fabric : ${modName}(), ModInitializer {

    override fun onInitialize() {
    }

    override fun isClothConfigLoaded(): Boolean =
        FabricLoader.getInstance().isModLoaded("cloth-config2")

    override fun getConfigScreen(parent: Screen): Screen =
        AutoConfig.getConfigScreen(ModConfigData::class.java, parent).get()
}
